test/e2e-go/features/transactions/onlineStatusChange_test.go

```diff
@@ -171,7 +172,11 @@ func TestCloseOnError(t *testing.T) {
 	_, curRound := fixture.GetBalanceAndRound(initiallyOnline)
 
 	var partkeyFile string
-	_, partkeyFile, err = client.GenParticipationKeysTo(initiallyOffline, 0, curRound+1000, 0, t.TempDir())
+	installFunc := func(keyPath string) error {
+		_, err := client.AddParticipationKey(keyPath)
+		return err
+	}
+	_, partkeyFile, err = participation.GenParticipationKeysTo(initiallyOffline, 0, curRound+1000, 0, t.TempDir(), installFunc)
```

Comment:
I'm a bit confused about usage of `participation.GenParticipationKeysTo` when both `outDir` and an install function are provided. Based on my reading of the code, the install function won't be called if `outDir` is provided, but nearly all instances provide both arguments

Reply:
If the install function isn't used/needed, I'd prefer callers pass in `nil` for simplicity/clarity.

Reply:
And I'd maybe also advocate for the function to error if both `outDir` and an install function are provided. Otherwise it seems too easy to mess up

Reply:
Added an early error and test case for this bad input condition.

Reply:
Also modified the install functions to return an error. This is a stronger assertion since install should not be called.